### PR TITLE
🏗 test-status app refactors + report e2e tests

### PR DESCRIPTION
**Pull request by danielrozenberg**
* Replace the test-type inference code from `status-report.js` with an explicit `--test-status-type` flag
* Report e2e test runs to the test-status app #e2e

### CODE REVIEW

File: build-system/tasks/e2e/index.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const Mocha = require('mocha');
 const tryConnect = require('try-net-connect');
 const {cyan} = require('ansi-colors');
 const {execOrDie, execScriptAsync} = require('../../exec');
+const {reportTestStarted} = require('../runtime-test/status-report');
```
Comment (estherkim) [inline build-system/tasks/e2e/index.js]: nit: move `status-report.js` file out of `tasks/runtime-test`

Comment (danielrozenberg) [reply build-system/tasks/e2e/index.js]: Done

**Comment by danielrozenberg** [comment]
Discussed offline, reverting to the original type+subtype inference code. If and when we refactor `gulp test` to a split of `gulp unit` and `gulp integration` we can take a look at how we change this

File: build-system/tasks/report-test-status.js [hunk 2]
```diff
@@ -34,37 +37,42 @@ const TEST_TYPE_SUBTYPES = new Map([
   // TODO(danielrozenberg): add 'saucelabs' to integration tests when supported.
   ['integration', ['local', 'single-pass']],
   ['unit', ['local', 'local-changes', 'saucelabs']],
-  // TODO(danielrozenberg): add 'e2e' tests.
+  ['e2e', ['local']],
 ]);
 const TEST_TYPE_BUILD_TARGETS = new Map([
   ['integration', ['RUNTIME', 'FLAG_CONFIG', 'INTEGRATION_TEST']],
   ['unit', ['RUNTIME', 'UNIT_TEST']],
```
Comment (rsimha) [inline build-system/tasks/report-test-status.js]: `['e2e', ['RUNTIME', 'FLAG_CONFIG', 'E2E_TEST']],`

Comment (danielrozenberg) [reply build-system/tasks/report-test-status.js]: The sins of rebasing and removing commits >__<

Done.

**Comment by danielrozenberg** [comment]
Let's do this in a separate PR. I still have a few holes in my knowledge when it comes to when asyncs/awaits apply

**Pull request merged**